subsys/bluetooth/host/id.c

```diff
@@ -775,7 +775,10 @@ void bt_id_add(struct bt_keys *keys)
 
 	err = hci_id_add(keys->id, &keys->addr, keys->irk.val);
 	if (err) {
-		BT_ERR("Failed to add IRK to controller");
+		/* Duplicate device is OK */
+		if (err != BT_HCI_ERR_INVALID_PARAM) {
```

Comment:
Although the HCI specification does not list other cases where BT_HCI_ERR_INVALID_PARAM can be returned, I'm a bit reluctant to accept all cases of BT_HCI_ERR_INVALID_PARAM here.